tests/e2e/utils.js

```diff
@@ -336,3 +342,46 @@ export function useTheme( themeSlug ) {
 		await activateTheme( previousTheme );
 	} );
 }
+
+/**
+ * Creates a basic coupon with the provided coupon amount. Returns the coupon code.
+ *
+ * @param {Object} [coupon] Coupon object. Default to fixed cart type and amount = 5.
+ * @param {string}  [coupon.amount] Amount to be applied. Defaults to 5.
+ * @param {string} [coupon.discountType] Type of a coupon. Defaults to Fixed cart discount.
+ * @param {number} [coupon.usageLimit] How many times the coupon can be used in total. Defaults to -1.
+ */
+export const createCoupon = async ( coupon ) => {
```

Comment:
I wonder if moving this function to https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/tests/utils/merchant.js would make more sense, as it's something the merchant would do. Similar to having `applyCouponFromCheckout()` located in https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/tests/utils/shopper.js.

Reply:
I thought about that too! But, from what I understood based on our file structure and Woo's as well, the functions that we add to the `merchant` should describe its flow (i.e., Describing how a merchant would take a specific action on the admin page). And the `createCoupon` function uses Rest API instead. So, this isn't how a merchant would create a coupon.

Reply:
Ah, right. That makes sense. I haven't seen it from that angle. Thanks for explaining that, @tarhi-saad.